src/lib/sidenav/sidenav.ts

```diff
@@ -161,7 +162,7 @@ export class MdSidenav implements AfterContentInit, OnDestroy {
     let activeEl = this._doc && this._doc.activeElement;
     if (activeEl && this._elementRef.nativeElement.contains(activeEl)) {
       if (this._elementFocusedBeforeSidenavWasOpened instanceof HTMLElement) {
-        this._elementFocusedBeforeSidenavWasOpened.focus();
+        this._focusOriginMonitor.focusVia(this._elementFocusedBeforeSidenavWasOpened, 'program');
```

Comment:
That makes sense. I was trying to think of something that would work automagically.